test/test_taxonomy.py

```diff
@@ -506,6 +507,92 @@ def testHIV(self):
         )
 
 
+class TestAllowedTaxonomicRank(TestCase):
+    """
+    Test the isAllowedTaxonomicRank function.
+    """
+
+    def testEmpty(self):
+        """
+        The function must return True when more than one part of the lineage matches.
+        """
+        lineage = (
+            LE(245, "x", "realm"),
+            LE(246, "bacteria", "kingdom"),
+            LE(247, "y", "species"),
+        )
+        self.assertTrue(
+            isAllowedTaxonomicRank(
+                set((("bacteria", "kingdom"), ("y", "species"))), lineage
+            )
+        )
+
+    def testExactCase(self):
+        """
+        The function must return True when case matches exactly.
+        """
+        lineage = (
+            LE(245, "x", "realm"),
+            LE(246, "bacteria", "kingdom"),
+            LE(247, "y", "species"),
+        )
+        self.assertTrue(
+            isAllowedTaxonomicRank(set((("bacteria", "kingdom"),)), lineage)
+        )
+
+    def testNonMatchingCase(self):
+        """
+        The function must return True when case doesn't match.
+        """
+        lineage = (
+            LE(245, "x", "realm"),
+            LE(246, "bacteria", "kingdom"),
+            LE(247, "y", "species"),
+        )
+        self.assertTrue(
+            isAllowedTaxonomicRank(set((("BACTERIA", "KINGDOM"),)), lineage)
+        )
+
+    def testMultipleMatches(self):
+        """
+        The function must return True when more than one part of the lineage matches.
+        """
+        lineage = (
+            LE(245, "x", "realm"),
+            LE(246, "bacteria", "kingdom"),
+            LE(247, "y", "species"),
+        )
+        self.assertTrue(
+            isAllowedTaxonomicRank(
+                set((("bacteria", "kingdom"), ("y", "species"))), lineage
+            )
+        )
+
+    def testRankMismatch(self):
+        """
+        The function must return False when the name matches but the rank does not.
+        """
+        lineage = (
+            LE(245, "x", "realm"),
+            LE(246, "bacteria", "kingdom"),
+            LE(247, "y", "species"),
+        )
+        self.assertFalse(
+            isAllowedTaxonomicRank(set((("bacteria", "species"),)), lineage)
+        )
+
+    def testNameMismatch(self):
+        """
+        The function must return False when the rank matches but the name does not.
+        """
+        lineage = (
+            LE(245, "x", "realm"),
+            LE(246, "bacteria", "kingdom"),
+            LE(247, "y", "species"),
+        )
+        self.assertFalse(isAllowedTaxonomicRank(set((("virus", "kingdom"),)), lineage))
+
+
```

Comment:
I guess you could test False when rank and name don't match.

Reply:
I thought about it, but the two prior tests show that you get `False` back if even one of them doesn't match.  But..... actually, you're right - not testing `False` when both mismatch is making an assumption about the implementation, which you're not supposed to do. Thanks :-)

Reply:
OK, I added a test (and fixed another that didn't match its docstring).